hera_qm/omnical_metrics.py

```diff
@@ -12,3 +28,796 @@ def get_omnical_metrics_dict():
+        omni_calfits : str
+            calfits file output from omnical, typically
+            ending in *.omni.calfits
+        """
+        # Get file info and other relevant metadata
+        self.filedir     = os.path.dirname(omni_calfits)
+        self.filestem    = '.'.join(os.path.basename(omni_calfits).split('.')[:-1])
+        self.filename    = os.path.basename(omni_calfits)
+        self.version_str = hera_qm_version_str
+        self.history     = ''
+        self.firstcal_file = None
+
+        # Instantiate Data Object
+        self.uv = UVCal()
+        self.uv.read_calfits(omni_calfits)
+
+        # Get relevant metadata
+        self.Nants  = self.uv.Nants_data
+        self.freqs  = self.uv.freq_array.squeeze()
+        self.Nfreqs = len(self.freqs)
+        self.jones  = self.uv.jones_array
+        self.pols   = np.array(map(lambda x: self.jones2pol[x], self.jones))
+        self.Npols  = self.uv.Njones
+        self.times  = self.uv.time_array
+        self.Ntimes = self.uv.Ntimes
+        self.ant_array = self.uv.ant_array
+
+        # Get omnical gains, move time axis in front of freq axis
+        self.omni_gains = np.moveaxis(self.uv.gain_array, 2, 3)[:, 0, :, :, :]
+
+        # Assign total chisq array
+        self.chisq = np.moveaxis(self.uv.quality_array, 2, 3)[:, 0, :, :, :]
+        self.chisq_tavg = np.median(self.chisq, axis=1)
+
+    def run_metrics(self, firstcal_file=None, cut_band=True, phs_noise_cut=1.0,
+                          phs_std_cut=0.3, chisq_std_cut=5.0):
+        """
+        run omnical metrics
+
+        Input:
+        ------
+        firstcal_file : str, default=None
+            path to a FirstCal *.calfits file
+            if fed, it will perform firstcal comparison metrics
+
+        cut_band : bool, defaul=True
+            cut bandpass edges before metrics.
+            if True, cuts 0:102 and 922:1024 channels from band
+
+        phs_noise_cut : float, default=1.0
+            cut in phase noise level w.r.t. frequency for
+            good/bad determination of phase solution.
+            see self.phs_FT_metric for details
+
+        phs_std_cut : float, default=0.3
+            cut in phase variability for
+            good/bad determination.
+            see self.phs_std_metric for details
+
+        chisq_std_cut : float, default=5.0
+            sets the cut in chisq variablity for good/bad determination.
+            the cut is the standard deviation of the chisq per-antenna times
+            chisq_std_cut, which means chisq_std_cut is like a "sigma" threshold
+            for each antenna's chisq variability.
+        """
+        # assign fc filename
+        self.firstcal_file = firstcal_file
+
+        # select freq channels
+        if cut_band == True:
+            self.band = np.arange(self.Nfreqs/10, self.Nfreqs - self.Nfreqs/10)
+        else:
+            self.band = np.arange(self.Nfreqs)
+
+        # Get robust standard deviation of chisq for each antenna
```

Comment:
Even though it's brief, you may want to break these calculations into a separate function like you do for the phase metrics. That will be more consistent, and making testing a lot easier.